app/Http/Controllers/Auth/RegisterController.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Http\Controllers\Auth;
+
+use App\Models\User;
+use App\Http\Controllers\Controller;
+use App\Http\Requests\RegisterUserRequest;
+use Illuminate\Contracts\View\Factory;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Facades\Hash;
+
+class RegisterController extends Controller
+{
+    /**
+     * Register a new user
+     *
+     * @param RegisterUserRequest $request
+     * @return Factory|View|\Illuminate\Http\RedirectResponse
+     */
+    public function register(RegisterUserRequest $request)
+    {
+        $user = User::create([
+            'username' => $request->username,
```

Comment:
we can take only validated input like `$allDetails = $request->validated()` and instead of taking value of $request object each time we can take value from $allDetails array